src/app/app.ts

```diff
@@ -0,0 +1,39 @@
+import {Post} from 'mattermost-redux/types/posts';
+import {AppCall, AppContext} from 'mattermost-redux/types/apps';
+
+import {ENV} from '../utils';
+
+import {mattermost, zendesk} from '../clients';
+
+import store from '../store/config';
+
+import {getTicketForPost} from './model';
+
+class App {
+    createTicketFromPost = async (call: AppCall): Promise<string> => {
+        const ticket = getTicketForPost(call.values);
+
+        const zdClient = zendesk.newClient(ENV.zendesk.username, ENV.zendesk.apiToken, ENV.zendesk.apiURL);
+
+        const result = await zdClient.tickets.create(ticket);
+        const user = await zdClient.users.show(result.requester_id);
```

Comment:
Maybe surround these two calls with `try/catch`. Return an error like `Failed to create ticket` etc. Unless the zendesk client is already returning an error like that

Reply:
I think adding context from the app would be beneficial.  Zendesk is returning an error, it is somewhat cryptic and it would be nice to understand what part of the code failed.  

How about the following prepended message? 
![image](https://user-images.githubusercontent.com/7575921/103679594-a3403380-4f4a-11eb-8f8f-048c1c04d3da.png)

Reply:
Note that I also now catch the error when failing to post a bot post to mattermost and added that to the modal
![image](https://user-images.githubusercontent.com/7575921/103680936-7a20a280-4f4c-11eb-939d-6e0e6939f438.png)

Reply:
Thanks for this suggestion :)